web/src/features/charts/bar-breakdown/BarBreakdownChart.tsx

```diff
@@ -84,6 +85,17 @@ function BarBreakdownChart({
     setTooltipData(null);
   };
 
+  const emissionData = [
+    ...new Set(
+      [
+        ...Object.values(currentZoneDetail?.dischargeCo2IntensitySources || {}),
+        ...Object.values(currentZoneDetail?.productionCo2IntensitySources || {}),
+      ].flatMap((item) => item.split('; '))
+    ),
+  ]
+    .filter((item) => !item.startsWith('assumes'))
+    .sort();
```

Comment:
This is going to be a very costly operation performance wise but there is a few ways we can make it faster. Will try a few things locally over easter and give some more concrete suggestions.